packages/ui/src/components/va-select/VaSelectOptionList/VaSelectOptionList.vue

```diff
@@ -95,143 +70,180 @@ const SelectOptionListPropsMixin = Vue.with(SelectOptionListProps)
     'no-previous-option-to-hover',
     'scroll-bottom',
   ],
-})
-export default class VaSelectOptionList extends mixins(
-  ColorMixin,
-  SelectOptionListPropsMixin,
-) {
-  itemRefs: Record<number, any> = {}
-
-  created () {
-    watch(() => this.$props.hoveredOption, (newOption: any) => {
-      if (newOption) {
-        this.scrollToOption(newOption)
+  props: {
+    ...useColorProps,
+    options: { type: Array as PropType<any[]>, default: () => [] },
```

Comment:
Let's have a SelectOption type.

```suggestion
    options: { type: Array as PropType<SelectOption[]>, default: () => [] },
```

Reply:
Aren't options can be literally of any type?

Reply:
It's actually `stringOrNumber | Record<string, string>`. Anyway typing SelectOption would make code easier to understand, for example you can use this type in hoveredOption instead of repeating `null | string | number | Record<string, any>` `selectOption | null`. This will not impact performance so it;s just nice to have.

Reply:
To add up, select generally functions as `Select<T>`, where `T` is option type. So having same type for all options is pretty important.